modules/navigation_3d/3d/nav_map_builder_3d.cpp

```diff
@@ -336,7 +333,9 @@ void NavMapBuilder3D::_build_step_navlink_connections(NavMapIterationBuild3D &r_
 
 		// If we have both a start and end point, then create a synthetic polygon to route through.
 		if (closest_start_polygon && closest_end_polygon) {
-			Polygon &new_polygon = link_polygons[link_poly_idx++];
+			link.navmesh_polygons.clear();
```

Comment:
clear is a resize(0), so this is not needed as you resize 1 below.

Reply:
If I just resize(1) when the link already has one polygon it keeps part of the old stuff as I am only updating id and owner here. This may not affect this PR as I reset the polygons in another part but in a branch where this did no longer happen this was actually a bug without the clear.